File: knowledge/src/main/java/com/ayvytr/knowledge/contract/GankHistoryContract.java
package com.ayvytr.knowledge.contract;

import com.ayvytr.commonlibrary.bean.GankHistory;
import com.ayvytr.mvpbase.IModel;
import com.ayvytr.mvpbase.IView;

import java.util.List;

import io.reactivex.Observable;

/**
 * @author admin
 */
public class GankHistoryContract {
    public interface Model extends IModel {
        Observable<GankHistory> getGankHistory();
    }

    public interface View extends IView {
        void showGankHistory(List<String> results);
    }
}

File: knowledge/src/main/java/com/ayvytr/knowledge/contract/GankHistoryContract.kt
package com.ayvytr.knowledge.contract

import com.ayvytr.commonlibrary.bean.GankHistory
import com.ayvytr.mvp.IModel
import com.ayvytr.mvp.IView

import io.reactivex.Observable

/**
 * @author admin
 */
class GankHistoryContract {
    interface Model : IModel {
        val gankHistory: Observable<GankHistory>
    }

    interface View : IView {
        fun showGankHistory(results: List<String>)
    }
}
